### PR TITLE
Add inbound connections to client pool

### LINKED ISSUE #318: Add inbound connections to client pool
`BamClientPool`: https://github.com/tenx-tech/swap/blob/3d3bd5816ddb42108baf291977de06f59dfbc417/application/comit_node/src/comit_client/bam.rs

Is meant to be the source of connections for other nodes. However if you get connected to by another party and then want to send them a message you will have to create a new connection because the existing one is not stored.

When the server receives a new connection it should add the client to `BamClientPool` under its `SocketAddress` and we should index the clients by socket address (for now). 

DoD:
- [x] `GET /peers` on Node2 returns Node1 network details once Node1 has sent a SWAP request to Node2
- [x] ~~`GET /peers` on Node1 and Node2 returns only node network details once Node2 has sent a SWAP request to Node1 (this goes after test above)~~

Blocked by #555 
Blocked by #620 
~~Blocked by Noise and Pubkeys (@LLFourn to open Noise ticket)~~

Note:

Ensuring that an old connection can be reused is out of scope for this ticket, because it requires connections to be uniquely identified.

### PR DESCRIPTION
**Pull request by luckysori**
Resolves #318

---
- [x] RTM (Ready To Merge)

### CODE REVIEW

**Comment by D4nte** [comment]
Please update title of #312 accordingly and ensure @LLFourn has done his todo :)

File: api_tests/dry/rfc003/test.js [hunk 1]
```diff
@@ -140,7 +149,7 @@ describe("RFC003 HTTP API", () => {
             });
     });
 
-    it("[Alice] Should see Bob's IP in her list of peers after sending a swap request to him", async () => {
+    it("[Alice] Should see Bob in her list of peers after sending a swap request to him", async () => {
```
[the remark this answers is not in the document]
Comment (luckysori) [reply api_tests/dry/rfc003/test.js]: Good point!

**Comment by D4nte** [comment]
Does the title still apply? Are we only pooling inbound connections? I'd expect to pull outbound connections too.

**Comment by luckysori** [comment]
> Looking at how small the code change is I am not sure to understand why #630 was split in two. This could have just been parked for until noise is done.

The issue was added to the sprint backlog so I thought it was ready to be implemented.

**Comment by luckysori** [comment]
> Does the title still apply? Are we only pooling inbound connections? I'd expect to pull outbound connections too.

Outbound connections were already being pooled prior to this PR.

**Comment by luckysori** [comment]
> Please update title of #318 accordingly and ensure @LLFourn has done his todo :)

Now done, epic tracking comit node public keys [here](https://github.com/comit-network/comit-rs/issues/637).

File: application/comit_node/src/comit_server.rs [hunk 2]
```diff
@@ -18,7 +26,13 @@ pub fn listen<B: BobSpawner>(
         let config = swap_config(Arc::clone(&bob_spawner));
 
         let connection = Connection::new(config, codec, connection);
-        let (close_future, _client) = connection.start::<json::JsonFrameHandler>();
+        let (close_future, client) = connection.start::<json::JsonFrameHandler>();
+
+        if let Ok(addr) = peer_addr {
+            let bam_client = Arc::new(BamClient::new(addr, client));
+            debug!("Adding {:?} to list of peers", addr);
```
Comment (thomaseizinger) [inline application/comit_node/src/comit_server.rs]: :x: I think this log statement should be in the implementation.

File: application/comit_node/src/comit_client/mod.rs [hunk 3]
```diff
@@ -25,6 +25,7 @@ pub trait Client: Send + Sync + 'static {
 
 pub trait ClientFactory<C>: Send + Sync + Debug {
     fn client_for(&self, comit_node_socket_addr: SocketAddr) -> Result<Arc<C>, ClientFactoryError>;
+    fn add_client(&self, comit_node_socket_addr: SocketAddr, client: Arc<C>);
```
Comment (LLFourn) [inline application/comit_node/src/comit_client/mod.rs]: This should have just been a function on BamClientPool IMO.

Comment (D4nte) [reply application/comit_node/src/comit_client/mod.rs]: @LLFourn we can correct with #651

**Pull request merged**